rmw_fastrtps_cpp/src/type_support_common.cpp

```diff
@@ -47,6 +47,7 @@ void TypeSupport::set_members(const message_type_support_callbacks_t * members)
 
   // Total size is encapsulation size + data size
   m_typeSize = 4 + data_size;
+  m_typeSize = (m_typeSize + 3) & ~3;
```

Comment:
@MiguelCompany mind to explain this change?

Reply:
@hidmic 
**TL/DR:** Account for RTPS sub-message alignment

The DDSI-RTPS spec requires each submessage to be aligned on a 4-byte boundary. This means that, when using `PREALLOCATED` memory policy, we could receive a serialized payload bigger than `m_typeSize`, which would thus be discarded. When using the rmw's default `PREALLOCATED_WITH_REALLOC` memory policy, an unnecessary reallocation could happen.

Reply:
I added some comments on f40ff2d

Reply:
I see. How is this not a problem on `master`? Upstream Fast-DDS differences?

Reply:
@hidmic You are right! It is a problem on `master`. I've created #550 with the proper fix.